lib/Cro/HTTP/Router.pm6

```diff
@@ -1045,4 +1045,36 @@ module Cro::HTTP::Router {
             }
         }
     }
+    
+    sub static-resource(Str() $base, *@path, :$mime-types, :@indexes) is export {
+        my $resp = $*CRO-ROUTER-RESPONSE //
+        die X::Cro::HTTP::Router::OnlyInHandler.new(:what<route>);
+
+        my $path = ($base, |@path).join: '/';
+        my %fallback = $mime-types // {};
+
+        sub get-mime($ext) {
+            %mime{$ext} // %fallback{$ext} // 'application/octet-stream';
+        }
+
+        sub get-extension($path) {
+            return ($path ~~ m/ '.' ( <-[ \. ]>+ ) $ / )[0].Str;
+        }
+
+        if $path and my $resource = %?RESOURCES{$path} and $resource.IO.e {
+            content get-mime(get-extension($path)), slurp($resource, :bin);
+        } else {
+            for @indexes {
+                my $index = ($path, $_).grep(*.so).join: '/';
+                my $resource = %?RESOURCES{$index};
+                if $resource.IO.e {
+                    content get-mime(get-extension($index)), slurp($resource, :bin);
+                    return;
+                }
+            }
+        }
+
+        $resp.status = 404;
```

Comment:
This sets `404` unconditionally even if we found the resource, since there's no `return` in the first path of the `if` statement above. But I'd probably toss the `return` (make it `last`) and then make this `$resp.status //= 404`.

Reply:
done?!